File: src/App/DTO/FeedbackAuthorDTO.php
<?php
declare(strict_types=1);

namespace App\DTO;

use Articus\DataTransfer\Annotation as DTA;

/**
 * Информация об авторе отзыва.  Если отзыв оставлен анонимно, параметр не возвращается.
 */
class FeedbackAuthorDTO
{
    /**
     * Имя автора отзыва.
     * @DTA\Data(field="name", nullable=true)
     * @DTA\Validator(name="Scalar", options={"type":"string"})
     */
    public ?string $name = null;

    /**
     * @DTA\Data(field="region", nullable=true)
     * @DTA\Strategy(name="Object", options={"type":\App\DTO\RegionDTO::class})
     * @DTA\Validator(name="TypeCompliant", options={"type":\App\DTO\RegionDTO::class})
     */
    public ?\App\DTO\RegionDTO $region = null;

}

File: src/main/kotlin/ympa_kotlin_wiremock_server/models/FeedbackAuthorDTO.kt
@file:Suppress(
    "RemoveRedundantQualifierName",
    "unused",
)

package ympa_kotlin_wiremock_server.models

import com.fasterxml.jackson.annotation.JsonProperty

data class FeedbackAuthorDTO(
    @field:JsonProperty("name")
    val name: kotlin.String? = null,

    @field:JsonProperty("region")
    val region: RegionDTO? = null,

)
